wordpress-mobile/WordPress-iOS
# Reader: Use performBatchUpdates instead of beginUpdates endUpdates.

Refs #11113 

This PR is a guess at a fix for #11113.  

In the issue, @jkmassel posits a fix might involve toggling `tableViewHandler setListensForContentChanges:` before and after the `beginUpdates/endUpdates` calls.  Its a good thought but `ReaderCommentsViewController` actually does not listen for content changes and already sets this to `NO` when [configuring the tableview handler](https://github.com/wordpress-mobile/WordPress-iOS/blob/11.8/WordPress/Classes/ViewRelated/Reader/ReaderCommentsViewController.m#L335).

We currently have three reports of this crash during the past 90 days and all three have happened since Feb 13. All three are running iOS 12.1.4.  This is too small a sample size to say for sure but its interesting that they've all happened recently and on the same version of the OS starting just a few days after it was released.  Stack traces are not identical and we see the crash occurs in three difference scenarios:
- once when assigning content to a cell's WPRichContentView
- once when refreshing cell layout after scrolling had stopped
- once when responding to a delegate method that a cell's layout needed to be updated.

This is actually helpful in eliminating potential causes (like a conflict with an existing inflight animation block) even if it doesn't identify the actual cause. 

So, a bit of history, the calls to `beginUpdates` and `endUpdates` are made in order to resize cells after embedded media has been loaded in a way that avoids reloading the cell.  Its a technique actually [mentioned in the docs](https://developer.apple.com/documentation/uikit/uitableview/1614908-beginupdates). 

The only theory I have regarding the cause is that calling `beginUpdates/endUpdates` kicks off an animation block and while that block is inflight more content is fetched and the table view is reloaded before the animation completes.  The docs say to avoid calling `tableView reloadData` between `beginUpdates/endUpdates`. It seems possible calling it while an animation is running would also be an issue.  I tried testing this theory but inserting a new comment and reloading the tableview right after the call to `endUpdates`. I could spam the comment list with more and more comments this way but it still didn't trigger the crash. :( 

With iOS 11, we have some new, recommended API that we might use to the same affect and see if it avoids the crash.   This PR checks for the availability of that API and will call `performBatchUpdates:completion:` when possible.

To test:
View comments in the reader.  Find a post with a variety of text only comments and comments with embedded images or video.   I used [this one](https://aerychtest.wordpress.com/2014/10/24/a-post-for-testing-images-and-media-in-comments/). 
Confirm that cells with images and video are sized correctly with this change. 

@jklausa could I trouble you for a review of this one?
